pylds/base.py

```diff
@@ -102,131 +130,61 @@ def vector_field_flat(t, points, vector_field, box_BOUNDARIES):
     """
     Returns the values of the LD function from integrated trajectories from initial conditions in phase-space.
     
     Parameters
     ----------
-    points_initial : ndarray, shape(n, )
-        initial conditions in phase-space.
+    grid_parameters : list of 3-tuples of floats
+        input parameters of limits and size of mesh per axis
     
     vector_field: function
         vector field over phase-space
         
-    LD_PARAMETERS : list made of a 3-tuple and a float
-        input parameters for LD computation
-        3-tuple contains floats t_initial, t_final, dt (timestep)
-        float is p-value of Lp-norm.
+    tau : float
+        Upper limit of integration.
+        
+    p_value : float, optional
+        Exponent in Lagrangian descriptor definition. The default is 0.5.
     
-    box_BOUNDARIES : list of 2-tuples, optional
-        box boundaries for escape condition of variable time integration
-        boundaries are infinite by default.
+    box_boundaries : list of 2-tuples, optional
+        Box boundaries for escape condition of variable time integration.
+        Boundaries are infinite by default.
     
     Returns
     -------
-    LD : 1d array, shape (n, )
-        array of computed LD values for all initial conditions.
+    LD : ndarray, shape (Nx, Ny)
+        Array of computed Lagrangian descriptor values for all initial conditions.
     """
-    f = vector_field_flat
     
-    t_initial, t_final, dt = LD_PARAMETERS[0]
-    p_norm = LD_PARAMETERS[1]
+    f = lambda x, y: vector_field_flat(x, y, vector_field, p_value, box_boundaries)
+    y0 = generate_points(grid_parameters)
     
-    if (t_final - t_initial)*dt < 0:
-        dt = -dt
-        
-    shape = points_initial.shape
-    LD = np.zeros(shape[0])  # Array to store forward LDs
-    for t in np.arange(t_initial, t_final - dt, dt):
-        ###################################
-        # Inputs for integrator
-        t_span = (t, t + dt)
-        y0 = points_initial.flatten()
-        ###################################
-        # Outputs from integration
-        solution_object = integrator(f, t_span, y0, args=(vector_field, box_BOUNDARIES))
-        points_final = extract_y_final(solution_object)
-        points_final = points_final.reshape(shape)
-        ###################################
-        # Compute LD
-        LD = accumulate_lagrangian_descriptor(LD, points_initial, points_final, LD_PARAMETERS)
-        ###################################
-        # For next iteration
-        points_initial = points_final
+    solution = solve_ivp(f, [0,tau], y0, t_eval=[tau], rtol=1.0e-4)
     
-    return LD
-
+    n=len(grid_parameters)
+    LD_values = solution.y[n::n+1] #values corresponding to LD
+    
+    LD=np.abs(LD_values).reshape((grid_parameters[0][2],grid_parameters[1][2])).T #reshape to 2D array
```

Comment:
Instead of using `(grid_parameters[0][2],grid_parameters[1][2])` to pass the shape, unpack paramters in a line above instead to make explicit what specific parameters are being used for reshaping.

Reply:
Done